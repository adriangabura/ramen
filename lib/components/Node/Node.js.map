{"version":3,"file":"Node.js","sourceRoot":"","sources":["../../../src/components/Node/Node.tsx"],"names":[],"mappings":";;;;;;;;;AAAA,2CAA+B;AAE/B,SAAS,IAAI;IACX,OAAO,CAAC,wCAAe,CAAC,CAAC;AAC3B,CAAC;AAED,kBAAe,IAAI,CAAC"}