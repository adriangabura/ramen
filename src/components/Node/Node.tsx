import React from "react";
import Draggable from "react-draggable";
import styled from "styled-components";

import { lightTheme } from "../../themes";
import { IGraphNodeProps } from "../../types";

const NodeContainer: any = styled.div`
  z-index: 100;
  position: absolute;
  display: inline-block;
  width: ${(props: IGraphNodeProps) => props.width}px;
  color: ${({ theme }) => theme.textColor || lightTheme.textColor};
  background: ${({ theme }) => theme.nodeBackground || lightTheme.nodeBackground};
  border-radius: ${({ theme }) => theme.borderRadius || lightTheme.borderRadius};
  min-height: 108px;
  display: inline-block;
  cursor: move;
  border: ${({ theme }) => theme.borderWidth || lightTheme.borderWidth} solid ${({ theme }) => theme.borderColor || lightTheme.borderColor};
`;

const NodeTitle = styled.div`
  padding: .5rem .8rem;
  height: 36px;
`;

function Node(props: IGraphNodeProps) {
  const { children, x = 0, y = 0, width = 200, name, zoomLevel = 1 } = props;
  const { onDrop = () => { }, onDrag = () => { } } = props;

  return (
    <Draggable
      scale={zoomLevel}
      bounds="#GraphEditor"
      defaultPosition={{ x, y }}
      cancel=".noDrag"
      onDrag={onDrag}
      onStop={onDrop}
    >
      <NodeContainer width={width}>
        <NodeTitle>
          {name}
        </NodeTitle>
        <div>
          {children}
        </div>
      </NodeContainer>
    </Draggable>
  );
}

export default Node;
