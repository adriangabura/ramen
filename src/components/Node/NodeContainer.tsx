import get from "lodash.get";
import React from "react";

import { RamenContext } from "../../context/RamenContext/RamenContext";
import DefaultField from "../Field/Field";
import DefaultNode from "./Node";

const PIN_SPACING = 42;

function NodeContainer(props: any) {
  const { node, Node = DefaultNode, Field = DefaultField, ...rest } = props;
  const { onFieldOutMouseDown, onFieldInMouseUp } = props;

  const { schema } = React.useContext(RamenContext);

  const outFields = get(schema, `nodeTypes.${node.type}.fields.out`, [])
    .map((field: any, idx: number) => (
      <Field
        color={get(schema, `socketTypes.${field.type}.color`)}
        key={idx}
        hasOutput={true}
        height={PIN_SPACING}
        onMouseDown={() => onFieldOutMouseDown(node.id, field.id)}
      >
        {field.name}
      </Field>
    ));

  const inFields = get(schema, `nodeTypes.${node.type}.fields.in`, [])
    .map((field: any, idx: number) => (
      <Field
        color={get(schema, `socketTypes.${field.type}.color`)}
        key={idx}
        hasInput={true}
        height={PIN_SPACING}
        onMouseUp={() => onFieldInMouseUp(node.id, field.id)}
      >
        {field.name}
      </Field>
    ));

  return (
    <Node {...rest} name={node.name || get(schema, `nodeTypes.${node.type}.name`, "")}>
      {outFields}
      {inFields}
    </Node>
  );
}

export default NodeContainer;
