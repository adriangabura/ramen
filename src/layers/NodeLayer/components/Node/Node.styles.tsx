import styled from "styled-components";

import { lightTheme } from "../../../../themes";

export const NodeWrapper: any = styled.div`
  z-index: 100;
  position: absolute;
  display: inline-block;
  width: ${({ width }: any) => width}px;
  color: ${({ theme }) => theme.textColor || lightTheme.textColor};
  background: ${({ theme }) => theme.nodeBackground || lightTheme.nodeBackground};
  border-radius: ${({ theme }) => theme.borderRadius || lightTheme.borderRadius};
  min-height: 108px;
  display: inline-block;
  cursor: move;
  border: ${({ theme }) => theme.borderWidth || lightTheme.borderWidth} solid ${({ theme }) => theme.borderColor || lightTheme.borderColor};
`;

export const NodeTitle = styled.div`
  padding: .5rem .8rem;
  font-size: 1.1rem;
  height: 36px;
`;

export const NodeSubtitle = styled.div`
  color: ${({ theme }) => theme.textSecondary || lightTheme.textSecondary};
  font-size: .9rem;
`;
