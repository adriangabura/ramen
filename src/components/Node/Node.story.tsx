import { storiesOf } from "@storybook/react";
import * as React from "react";

import Node from "./Node";

storiesOf("Node", module)
  .add("simple node", () => <Node />);
