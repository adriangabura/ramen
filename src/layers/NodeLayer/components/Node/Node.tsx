import get from "lodash.get";
import React from "react";

import { FIELD_HEIGHT, NODE_WIDTH } from "../../../../constants";
import { GraphContext } from "../../../../context/GraphProvider/GraphProvider";
import { INodeProps } from "../../../../types";
import DefaultField from "../Field/Field";
import { NodeTitle, NodeWrapper as DefaultNode } from "./Node.styles";

function Node(props: INodeProps) {
  const {
    node,
    CustomNode = DefaultNode,
    CustomField = DefaultField,
    className,
    onFieldOutMouseDown,
    onFieldInMouseUp,
    ...rest
  } = props;

  const { schema } = React.useContext(GraphContext);

  const outFields = get(schema, `nodeTypes.${node.type}.fields.out`, [])
    .map((field: any, idx: number) => (
      <CustomField
        color={get(schema, `socketTypes.${field.type}.color`)}
        key={idx}
        hasOutput={true}
        height={FIELD_HEIGHT}
        onMouseDown={() => onFieldOutMouseDown(node.id, field.id)}
      >
        {field.name}
      </CustomField>
    ));

  const inFields = get(schema, `nodeTypes.${node.type}.fields.in`, [])
    .map((field: any, idx: number) => (
      <CustomField
        color={get(schema, `socketTypes.${field.type}.color`)}
        key={idx}
        hasInput={true}
        height={FIELD_HEIGHT}
        onMouseUp={() => onFieldInMouseUp(node.id, field.id)}
      >
        {field.name}
      </CustomField>
    ));

  return (
    <CustomNode
      {...rest}
      id={node.id}
      width={NODE_WIDTH}
      className={"node " + className}
    >
      <NodeTitle>
        {node.name}
      </NodeTitle>
      <div>
        {outFields}
        {inFields}
      </div>
    </CustomNode>
  );
}

export default Node;
