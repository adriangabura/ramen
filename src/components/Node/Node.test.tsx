import React from "react";
import renderer from "react-test-renderer";

import Node from "./Node";

describe("Node", () => {

  it("renders without crashing", () => {
    const tree = renderer.create(<Node />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
