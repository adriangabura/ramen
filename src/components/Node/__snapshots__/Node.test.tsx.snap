// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node renders without crashing 1`] = `
<div>
  node
</div>
`;
